### PR TITLE
Update `pnpm` to `v9` & improve CI

### PR DESCRIPTION
**Pull request by xeho91**
Following the request: <https://github.com/storybookjs/addon-svelte-csf/pull/179#pullrequestreview-2053802933>

## Additional context

- Added `.nvmrc` _(I've used version that matched the previous CI workflows setup - `20`)_, so my machine can
automatically set up the right version.
- Upgraded CI workflows actions, and reduced one action - `pnpm/action-setup` -  in favor of `corepack`

<details>
  <summary>📦 Published PR as canary version: <code>4.1.3--canary.180.930cbf4.0</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @storybook/addon-svelte-csf@4.1.3--canary.180.930cbf4.0
  # or 
  yarn add @storybook/addon-svelte-csf@4.1.3--canary.180.930cbf4.0
  ```
</details>

### CODE REVIEW

**Comment by xeho91** [comment]
@JReinhold I'm puzzled how the `Chromatic` workflow was able to run successfully before, without the building step 🤔 👀

File: .github/workflows/chromatic.yml [hunk 1]
```diff
@@ -1,21 +1,35 @@
...
+
+      # https://pnpm.io/cli/install
+      - name: Install Node.js dependencies with pnpm
+        run: >
+          pnpm install
+          --frozen-lockfile
+
+      - name: Build package
+        run: >
+          pnpm run build
```
Comment (JReinhold) [inline .github/workflows/chromatic.yml:31]: Why is this necessary? I think that the addon is only using relative references in this repo, so building Storybook (which Chromatic always does) should just build the addon.

I'm fine with this step here, I'm just wondering if needing it is a symptom of something else I'm missing.

**Review by JReinhold — COMMENTED** [review]
Great! Only one question

File: .github/workflows/chromatic.yml (hunk 1 above)
Comment (xeho91) [reply .github/workflows/chromatic.yml:31]: It was failing, before I added this step. There's the log from the CI workflow: 
https://github.com/storybookjs/addon-svelte-csf/actions/runs/9073717361/job/24931260330#step:6:97

Comment (JReinhold) [reply .github/workflows/chromatic.yml:31]: Ah yes, probably because of this:

> WARN Could not resolve addon "../dist/preset/index.js", skipping. Is it installed?

I have no idea why this worked before..

**Review by JReinhold — APPROVED** [review]

**Comment by shilman** [comment]
:rocket: PR was released in [`v4.1.3`](https://github.com/storybookjs/addon-svelte-csf/releases/tag/v4.1.3) :rocket:
